provider/pkg/provider/provider.go

```diff
@@ -74,27 +75,59 @@ func loadMetadata(azureAPIResourcesBytes []byte) (*resources.CloudAPIMetadata, e
 }
 
 // Configure configures the resource provider with "globals" that control its behavior.
-func (k *googleCloudProvider) Configure(ctx context.Context,
+func (p *googleCloudProvider) Configure(ctx context.Context,
 	req *rpc.ConfigureRequest) (*rpc.ConfigureResponse, error) {
 	for key, val := range req.GetVariables() {
-		k.config[strings.TrimPrefix(key, "gcp-native:config:")] = val
+		p.config[strings.TrimPrefix(key, "gcp-native:config:")] = val
 	}
 
-	k.setLoggingContext(ctx)
+	p.setLoggingContext(ctx)
+
+	impersonateServiceAccountDelegatesString := p.getConfig("impersonateServiceAccountDelegates", "")
+	var impersonateServiceAccountDelegates []string
+	if impersonateServiceAccountDelegatesString != "" {
+		err := json.Unmarshal([]byte(impersonateServiceAccountDelegatesString), &impersonateServiceAccountDelegates)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to unmarshal '%s' as Impersonate Service Account Delegates", impersonateServiceAccountDelegatesString)
+		}
+	}
+
+	scopesString := p.getConfig("scopes", "")
+	var scopes []string
+	if scopesString != "" {
+		err := json.Unmarshal([]byte(scopesString), &scopes)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to unmarshal '%s' as Scopes", scopesString)
+		}
+	}
+
+	config := credentialsConfig{
+		Credentials:                        p.getConfig("credentials", "GOOGLE_CREDENTIALS"),
+		AccessToken:                        p.getConfig("accessToken", "GOOGLE_OAUTH_ACCESS_TOKEN"),
+		ImpersonateServiceAccount:          p.getConfig("impersonateServiceAccount", "GOOGLE_IMPERSONATE_SERVICE_ACCOUNT"),
+		ImpersonateServiceAccountDelegates: impersonateServiceAccountDelegates,
+		Scopes:                             scopes,
+	}
+
+	client, err := newGoogleHttpClient(ctx, config)
+	if err != nil {
+		return nil, err
+	}
+	p.client = client
 
 	return &rpc.ConfigureResponse{
 		AcceptSecrets: true,
 	}, nil
 }
 
 // Invoke dynamically executes a built-in function in the provider.
-func (k *googleCloudProvider) Invoke(_ context.Context, _ *rpc.InvokeRequest) (*rpc.InvokeResponse, error) {
+func (p *googleCloudProvider) Invoke(_ context.Context, _ *rpc.InvokeRequest) (*rpc.InvokeResponse, error) {
 	panic("Invoke not implemented")
```

Comment:
Should return a NYI error code rather than panicking here.
Example: https://github.com/pulumi/pulumi-azure-native/blob/1e4bfc64deec4e47f3dfd0d32ef7b66331ae9ef0/provider/pkg/provider/provider.go#L281